src/dbus_api/macros.rs

```diff
@@ -84,3 +84,14 @@ macro_rules! properties_footer {
         }
     };
 }
+
+macro_rules! pool_op_logging {
+    ($pre_oper:tt $(, $pre_args:expr)*; $post_oper:tt $(, $post_args:expr)*; $engine_op:expr) => {{
+        info!($pre_oper, $($pre_args),*);
+        let result = $engine_op;
+        if let Ok(ref uuid) = result {
+            info!($post_oper, $($post_args,)* uuid);
+        }
```

Comment:
I think it makes sense to also log a message at the warn level if the result is an error. Whether there is success or failure here, something will be returned across the D-Bus. If it is correct to log on a non-error return from the engine method, I think it makes sense to log on an error return as well.

Related, I'ld prefer if you leave out the word "succesfully" from the log messages. If the user will certainly see a warn message on a failure, "succesfully" is quite redundant.

Reply:
I don't know what to fill in for the rest of the warn message, but is this a good start?
```
        if let Ok(ref action) = result {
            info!($post_oper, $($post_args,)* action);
        } else {
            warn!("pool operation failed");
        }
```
Admittedly, I'm reluctant to add a third group of metavariables for error messaging.